aws-lc-rs/src/ec.rs

```diff
@@ -114,25 +115,40 @@ impl AlgorithmID {
     }
 }
 
+/// Elliptic curve public key.
 #[derive(Clone)]
-pub struct PublicKey(Box<[u8]>);
+pub struct PublicKey {
+    octets: Box<[u8]>,
+    der: Box<[u8]>,
+}
 
-impl Debug for PublicKey {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), fmt::Error> {
-        f.write_str(&format!("PublicKey(\"{}\")", test::to_hex(self.0.as_ref())))
-    }
+/// An elliptic curve public key as a DER-encoded `SubjectPublicKeyInfo` structure
+#[allow(clippy::module_name_repetitions)]
+pub struct EcPublicKeyDer {
+    _priv: (),
 }
 
 impl PublicKey {
-    fn new(pubkey_box: Box<[u8]>) -> Self {
-        PublicKey(pubkey_box)
+    /// Provides the public key as a DER-encoded `SubjectPublicKeyInfo` structure.
+    #[must_use]
+    pub fn as_der(&self) -> Buffer<'_, EcPublicKeyDer> {
```

Comment:
I'm really not to fond of how we are using the names `as_der` and `EcPublicKeyDer` write the output as DER following the X.509 ASN.1 `SubjectPublicKeyInfo` type defiition. DER is an ASN.1 encoding scheme, `SubjectPublicKeyInfo` is the structure to describe a public key that could be encoded in various formats, PEM, DER, BER, etc. I feel like we should be more explicit for own benefit in case we need to serialize to different structures.

Case in point, you could have a function that just writes the raw key bytes as a DER encode OCTET STRING. Just won't want overload where DER is so to speak.

Thoughts?

Reply:
Would `as_x509` and `X509PublicKey` any better?

As a reference, in the `BC-FIPS` Java world, the import of the der bytes (after base64 decoding if any) is something like:
```
new X509EncodedKeySpec(bytes)
```